plugins/premium/comments/comments_callback_mode.md

```diff
@@ -0,0 +1,596 @@
+  fetch(
+    'https://api.example/conversations/' + conversationUid + '/' + commentUid,
+    {
+      method: 'DELETE',
+    }
+  ).then(function (response) {
+    if (response.ok) {
+      done({ canDelete: true });
+    } else if (response.status === 403) {
+      done({ canDelete: false });
+    } else {
+      fail(new Error('Something has gone wrong...'));
+    }
+  });
+}
+
+tinymce.init({
+  selector: '#editor',
+  plugins: 'tinycomments',
+  tinycomments_mode: 'callback',
+  tinycomments_create: create_comment,
+  tinycomments_reply: reply_comment,
+  tinycomments_edit_comment: edit_comment,
+  tinycomments_delete: delete_comment_thread, // Add the callback to TinyMCE
+  tinycomments_delete_all: delete_all_comment_threads,
+  tinycomments_delete_comment: delete_comment,
+  tinycomments_lookup: lookup_comment
+});
+```
+
+### `tinycomments_delete`
+
+The `tinycomments_delete` function should asynchronously return a flag indicating whether the comment or comment thread was removed using the `done` callback. Unrecoverable errors are communicated to {{site.productname}} by calling the `fail` callback instead.
+
+The `tinycomments_delete` function is passed a (`req`) object as the first parameter, which contains the following key-value pair:
+
+`conversationUid`
+: The uid of the conversation the reply is targeting.
+
+The `done` callback should accept the following object:
+
+```js
+{
+  canDelete: boolean // whether or not the conversation can be deleted
+  reason: string? // an optional string explaining why the delete was not allowed (if canDelete is false)
+}
+```
+
+> **Note**: Failure to delete due to permissions or business rules is indicated by "false", while unexpected errors should be indicated using the "fail" callback.
+
+For example:
+
+```js
+function delete_comment_thread(_ref, done, fail) {
+  var conversationUid = _ref.conversationUid;
+  fetch('https://api.example/conversations/' + conversationUid, {
+    method: 'DELETE',
+  }).then(function (response) {
+    if (response.ok) {
+      done({ canDelete: true });
+    } else if (response.status === 403) {
+      done({ canDelete: false });
+    } else {
+      fail(new Error('Something has gone wrong...'));
+    }
+  });
+}
+
+tinymce.init({
+  selector: '#editor',
+  plugins: 'tinycomments',
+  tinycomments_mode: 'callback',
+  tinycomments_create: create_comment,
+  tinycomments_reply: reply_comment,
+  tinycomments_edit_comment: edit_comment,
+  tinycomments_delete: delete_comment_thread, // Add the callback to TinyMCE
+  tinycomments_delete_all: delete_all_comment_threads,
+  tinycomments_delete_comment: delete_comment,
+  tinycomments_lookup: lookup_comment
+});
+```
+
+### `tinycomments_delete_all`
+
+The `tinycomments_delete_all` function should asynchronously return a flag indicating whether all the comments in a conversation were removed using the `done` callback. Unrecoverable errors are communicated to {{site.productname}} by calling the `fail` callback instead.
```

Comment:
```suggestion
The `tinycomments_delete_all` function should asynchronously return a flag indicating whether all the comment threads were removed using the `done` callback. Unrecoverable errors are communicated to {{site.productname}} by calling the `fail` callback instead.
```
This is used to delete every thread in the document, not all comments in a thread/conversation (that's the tinycomments_delete callback).